sct-commons/src/main/java/org/lfenergy/compas/sct/commons/SclElementsProviderService.java

```diff
@@ -15,12 +16,24 @@
 import org.lfenergy.compas.sct.commons.scl.ied.*;
 import org.lfenergy.compas.sct.commons.scl.ldevice.LDeviceAdapter;
 import org.lfenergy.compas.sct.commons.scl.ln.AbstractLNAdapter;
+import org.lfenergy.compas.sct.commons.scl.ln.LnKey;
+import org.lfenergy.compas.sct.commons.util.Utils;
 
 import java.util.*;
+import java.util.function.BiPredicate;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class SclElementsProviderService implements SclElementsProvider {
```

Comment:
General question : how do we replace getDAI ?

Reply:
**getDAI** can be replaced by making call to **dataTypeTemplatesService**, then if we want to check also updatability from instances we call **lnService**

![image](https://github.com/com-pas/compas-sct/assets/76203423/d3d92c55-af64-47d2-bbc6-69e814c2d3e0)